include/picongpu/plugins/EnergyParticles.hpp

```diff
@@ -45,294 +46,415 @@
-    {
-        this->cellDescription = cellDescription;
-    }
+        /** method used by plugin controller to get --help description **/
+        void pluginRegisterHelp(boost::program_options::options_description& desc)
+        {
+            desc.add_options()
+                ((pluginPrefix + ".period").c_str(),
+                 boost::program_options::value<uint32_t > (&notifyPeriod),
+                 "compute kinetic and total energy [for each n-th step] enable plugin by setting a non-zero value");
+        }
 
-private:
+        //! method giving the plugin name (used by plugin control)
+        std::string pluginGetName() const
+        {
+            return pluginName;
+        }
 
-    /** method to initialize plugin output and variables **/
-    void pluginLoad()
-    {
-        if (notifyPeriod > 0) /* only if plugin is called at least once */
+      /** set cell description in this plugin **/
+        void setMappingDescription(MappingDesc *cellDescription)
         {
-            /* decide which MPI-rank writes output: */
-            writeToFile = reduce.hasResult(mpi::reduceMethods::Reduce());
+            this->cellDescription = cellDescription;
+        }
 
-            /* create two ints on gpu and host: */
-            gEnergy = new GridBuffer<float_64, DIM1 > (DataSpace<DIM1 > (2));
+    private:
 
-            if (writeToFile) /* only MPI rank that writes to file: */
+        /** method to initialize plugin output and variables **/
+        void pluginLoad()
+        {
+            if (notifyPeriod > 0) /* only if plugin is called at least once */
             {
-                /* open output file */
-                outFile.open(filename.c_str(), std::ofstream::out | std::ostream::trunc);
+                /* decide which MPI-rank writes output: */
+                writeToFile = reduce.hasResult(mpi::reduceMethods::Reduce());
+
+                /* create two ints on gpu and host: */
+                gEnergy = new GridBuffer<float_64, DIM1 > (DataSpace<DIM1 > (2));
 
-                /* error handling: */
-                if (!outFile)
+                if (writeToFile) /* only MPI rank that writes to file: */
                 {
-                    std::cerr << "Can't open file [" << filename
-                              << "] for output, diasble plugin output. " << std::endl;
-                    writeToFile = false;
+                    /* open output file */
+                    outFile.open(filename.c_str(), std::ofstream::out | std::ostream::trunc);
+
+                    /* error handling: */
+                    if (!outFile)
+                    {
+                        std::cerr << "Can't open file [" << filename
+                                  << "] for output, diasble plugin output. " << std::endl;
+                        writeToFile = false;
+                    }
+
+                    /* create header of the file */
+                    outFile << "#step Ekin_Joule E_Joule" << " \n";
```

Comment:
Should this code line executed if `outFile` is false? Maybe put an `else` before this line?